### PR TITLE
(feat)(Business Glossary) glossary term relationship graphql changes

### PR DESCRIPTION
**Pull request by saxo-lalrishav**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-web-react/src/graphql/glossaryTerm.graphql [hunk 1]
```diff
@@ -4,6 +4,28 @@ query getGlossaryTerm($urn: String!) {
...
+                    type    
+                }
+            }
+        }
+        downstreamLineage {
+            ...downstreamRelationshipFields
+        }
+        upstreamLineage {
+            ...upstreamRelationshipFields
+        }
```
Comment (gabe-lyons) [inline datahub-web-react/src/graphql/glossaryTerm.graphql:28]: We probably don't want to add downstreamLineage and upstreamLineage to the glossary term

Comment (saxo-lalrishav) [reply datahub-web-react/src/graphql/glossaryTerm.graphql:28]: yup, will remove it from here

**Comment by gabe-lyons** [comment]
Looking good- as discussed offline, we can move to use the new relationship type introduced here:

https://github.com/linkedin/datahub/pull/3094/files#diff-e04c9c2d80cbfd7aa7e3e0f867248464db0f6497684661132d6ead81ded21856R443

File: datahub-graphql-core/src/main/resources/gms.graphql [hunk 2]
```diff
@@ -940,6 +956,15 @@ type DataFlowDataJobsRelationships {
     entities: [EntityRelationshipLegacy]
 }
 
+type GlossaryTermIsARelationships {
+    entities: [EntityRelationshipLegacy]
+}
+
+type GlossaryTermHasARelationships {
+    entities: [EntityRelationshipLegacy]
+}
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/resources/gms.graphql:965]: we shouldn't need these types

File: datahub-web-react/src/graphql/glossaryTerm.graphql [hunk 3]
```diff
@@ -1,9 +1,33 @@
...
+            count
+            total
+            relationships {
+                entity {
+                    ... on GlossaryTerm {
+                        urn
+                    }
+                }
+            }
+        }
```
Comment (gabe-lyons) [inline datahub-web-react/src/graphql/glossaryTerm.graphql:30]: won't you also need incoming relationships?

**Review by gabe-lyons — COMMENTED** [review]
nice! will you also add the UI as part of this change?

File: datahub-web-react/src/graphql/glossaryTerm.graphql (hunk 3 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/graphql/glossaryTerm.graphql:30]: will take care this in UI PR as per requirement

**Comment by saxo-lalrishav** [comment]
Yes will raise a new PR for React

**Review by gabe-lyons — APPROVED** [review]
LGTM!

File: datahub-web-react/src/graphql/glossaryTerm.graphql [hunk 4]
```diff
@@ -1,9 +1,33 @@
-query getGlossaryTerm($urn: String!) {
+query getGlossaryTerm($urn: String!, $start: Int!, $count: Int!) {
```
Comment (gabe-lyons) [inline datahub-web-react/src/graphql/glossaryTerm.graphql:1]: hey lal- looks like adding these required params is causing build failures:

https://github.com/linkedin/datahub/pull/3103/checks?check_run_id=3353708342.

I would either remove them as required params (they are optional params to relationships() type) or populate them from react in this PR

Comment (saxo-lalrishav) [reply datahub-web-react/src/graphql/glossaryTerm.graphql]: yup, done

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**